src/libraries/System.Text.Json/gen/JsonSourceGenerator.cs

```diff
@@ -40,31 +40,32 @@ public void Initialize(GeneratorInitializationContext context)
         public void Execute(GeneratorExecutionContext executionContext)
         {
             SyntaxReceiver receiver = (SyntaxReceiver)executionContext.SyntaxReceiver;
-            List<CompilationUnitSyntax> compilationUnits = receiver.CompilationUnits;
-            if (compilationUnits == null)
+            List<ClassDeclarationSyntax>? contextClasses = receiver.ClassDeclarationSyntaxList;
+            if (contextClasses == null)
             {
                 return;
             }
 
             Parser parser = new(executionContext.Compilation);
-            _rootTypes = parser.GetRootSerializableTypes(receiver.CompilationUnits);
-
-            if (_rootTypes != null)
+            SourceGenerationSpec? spec = parser.GetGenerationSpec(receiver.ClassDeclarationSyntaxList);
+            if (spec != null)
             {
-                Emitter emitter = new(executionContext, _rootTypes);
+                _rootTypes = spec.ContextGenerationSpecList[0].RootSerializableTypes;
```

Comment:
The unit tests currently only specify one context class, so just the first is sufficient. I want to refactor these tests in a follow up so that we have a better way to validate correctness.